controllers/movies.ts

```diff
@@ -0,0 +1,302 @@
+      })) as Movie;
+      await Movie.remove(movie);
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async addMovie(params: MovieParams) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = await Movie.create({
+        Title: Title,
+        Year: Year,
+        Rated: Rated,
+        Released: Released,
+        Runtime: Runtime,
+        Genre: Genre,
+        Director: Director,
+        Writer: Writer,
+        Actors: Actors,
+        Plot: Plot,
+        Language: Language,
+        Country: Country,
+        Awards: Awards,
+        Poster: Poster,
+        Metascore: Metascore,
+        imdbRating: imdbRating,
+        imdbVotes: imdbVotes,
+        imdbID: imdbID,
+        Type: Type,
+        Response: Response,
+        Images: Images,
+        ComingSoon: ComingSoon,
+      });
+
+      await movie.save();
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async editMovie(params: MovieParams, id: any) {
+    const {
+      Title,
+      Year,
+      Rated,
+      Released,
+      Runtime,
+      Genre,
+      Director,
+      Writer,
+      Actors,
+      Plot,
+      Language,
+      Country,
+      Awards,
+      Poster,
+      Metascore,
+      imdbRating,
+      imdbVotes,
+      imdbID,
+      Type,
+      Response,
+      Images,
+      ComingSoon,
+    } = params;
+
+    try {
+      const movie = (await Movie.findOne({
+        where: {
+          Id: id,
+        },
+      })) as Movie;
+
+      movie.Title = Title;
+      movie.Year = Year;
+      movie.Rated = Rated;
+      movie.Released = Released;
+      movie.Runtime = Runtime;
+      movie.Genre = Genre;
+      movie.Director = Director;
+      movie.Writer = Writer;
+      movie.Actors = Actors;
+      movie.Plot = Plot;
+      movie.Language = Language;
+      movie.Country = Country;
+      movie.Awards = Awards;
+      movie.Poster = Poster;
+      movie.Metascore = Metascore;
+      movie.imdbRating = imdbRating;
+      movie.imdbVotes = imdbVotes;
+      movie.imdbID = imdbID;
+      movie.Type = Type;
+      movie.Response = Response;
+      movie.Images = Images;
+      movie.ComingSoon = ComingSoon;
+      await movie.save();
+      return movie;
+    } catch (err: any) {
+      throw new Error("Movie not found");
+    }
+  }
+
+  async getTotalLength() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+      let total = 0;
+      let minutes;
+      for (const movie of movies) {
+        if (movie.Runtime !== "N/A") {
+          if (movie.Type === "series") {
```

Comment:
Avoid deep nesting and multiple if statements deeply nested.

Reply:
I changed the logic, it is not nested anymore